packages/compass-shell/src/modules/runtime.js

```diff
@@ -70,6 +70,22 @@ function reduceSetupRuntime(state, action) {
     if (connection.options.checkServerIdentity === true) {
       delete connection.options.checkServerIdentity;
     }
+    // driver 4 doesn't support certificates as buffers, so let's copy paths
+    // back from model `driverOptions`
+    //
+    // TODO: Driver is not sure if buffer behavior was a bug or a feature,
+    // hopefully this can be removed eventually (see https://mongodb.slack.com/archives/C0V8RU15L/p1612347025017200)
+    ['sslCA', 'sslCRL', 'sslCert', 'sslKey'].forEach((key) => {
```

Comment:
do we need also the `tls` counterparts to be safe? or that's something compass just does not support?

Reply:
Compass doesn't seem to use them, yeah. I based this list on how `connection-model` picks up props to be loaded

https://github.com/mongodb-js/connection-model/blob/37bb63db26f384f393d0615c0436a48bd7383223/lib/connect.js#L33-L35

... but instead of this indexOf match, I picked up the list of props that are supposed to be files and start with ssl from driver 4:

https://github.com/mongodb/node-mongodb-native/blob/e5d981589d3413e4a240caaf11779748cfafe97c/src/connection_string.ts#L894-L917